server/iframe.go

```diff
@@ -222,14 +261,98 @@ func (a *API) authenticate(w http.ResponseWriter, r *http.Request) {
 	http.SetCookie(w, csrfCookie)
 
 	// Redirect to the home page
-	http.Redirect(w, r, "/", http.StatusSeeOther)
+	http.Redirect(w, r, redirectPath, http.StatusSeeOther)
+}
+
+// MessageHasBeenPosted is called when a message is posted in Mattermost. We rely on it to send a user activity notification
+// to Microsoft Teams when a user is mentioned in a message.
+// This is called in a controller Goroutine in the server side so there's no need to worry about concurrency here.
+func (p *Plugin) MessageHasBeenPosted(c *plugin.Context, post *model.Post) {
+	p.API.LogError("Message has been posted", "post_id", post.Id, "post_participants", post.Participants)
+
+	context := map[string]string{
+		"subEntityId": fmt.Sprintf("post_%s", post.Id),
+	}
+
+	jsonContext, err := json.Marshal(context)
+	if err != nil {
+		p.API.LogError("Failed to marshal context", "error", err.Error())
+		return
+	}
+
+	urlParams := url.Values{}
+	urlParams.Set("context", string(jsonContext))
+
+	for _, mention := range extractMentionsFromPost(post) {
+		u, err := p.apiClient.User.GetByUsername(mention)
+		if err != nil {
+			p.API.LogError("Failed to get user", "error", err.Error())
+			continue
+		}
+
+		msteamsUserID, exists := u.GetProp(getUserPropKey("user_id"))
+		if !exists {
+			p.API.LogError("MSTeams user ID is empty. Not sending notification.")
+			continue
+		}
+
+		appID, exists := u.GetProp(getUserPropKey("app_id"))
+		if !exists {
+			p.API.LogError("MSTeams app ID is empty. Not sending notification.")
+			continue
+		}
```

Comment:
Would we consider maintaining this mapping in our own table (or even just the kv store for the plugin) instead of putting it inside the user prop?

Reply:
@wiggin77 and I talked about this at some point and decided on the user properties since with one DB Call we had all the information for the user. This can even be improved to use `ListByUsernames` to get all the users at once.

Reply:
There is also the issue of cleanup when users are removed.